reco_utils/dataset/sparse.py

```diff
@@ -0,0 +1,204 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""
+Generate the user/item affinity matrix from a pandas dataframe and vice versa
+"""
+
+import pandas as pd
+
+import numpy as np
+import random
+import math
+
+import itertools
+
+from scipy.sparse import coo_matrix
+import logging
+
+# import default parameters
+from reco_utils.common.constants import (
+    DEFAULT_USER_COL,
+    DEFAULT_ITEM_COL,
+    DEFAULT_RATING_COL,
+    DEFAULT_TIMESTAMP_COL,
+    PREDICTION_COL,
+)
+
+# for logging
+log = logging.getLogger(__name__)
+
+
+class AffinityMatrix:
```

Comment:
Nice job! This looks similar to the reindexing in the SAR notebooks; maybe we could replace that too.

Reply:
yes, this is where I took it from :) I just thought that adding it into this function would have been easier for the user, as he/she doesn't need to implement it explicitly in the notebook.